src/main/scala/edu/ie3/util/osm/model/OsmEntity.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.util.osm.model
+
+import edu.ie3.util.geo.GeoUtils
+import edu.ie3.util.osm.model.OsmEntity.{MetaInformation, OsmEntityType}
+import edu.ie3.util.osm.model.OsmEntity.Relation.RelationMember
+import org.locationtech.jts.geom.Point
+
+import java.time.Instant
+import scala.annotation.tailrec
+
+sealed trait OsmEntity {
+  val osmModel: OsmEntityType
+  val id: Long
+  val tags: Map[String, String]
+  val metaInformation: Option[MetaInformation]
+
+  def hasKeyValuePair(key: CommonOsmKey, value: String): Boolean =
+    hasKeyValuePair(key.toString, value)
+
+  def hasKeyValuePair(key: String, value: String): Boolean =
+    tags.get(key).contains(value)
+
+  def hasKeyValuesPairOr(key: CommonOsmKey, values: Set[String]): Boolean =
+    hasKeyValuesPairOr(key.toString, values)
+
+  /** values.isEmpty equals to tagName=*
```

Comment:
This explanation might be a bit short. At least it wasn't understandable to me at the first sight. Also: Please complete the ScalaDoc